src/UI/class-admin-columns-parsely-stats.php

```diff
@@ -0,0 +1,370 @@
+ *   page_views: string,
+ *   visitors: string,
+ *   avg_time: string,
+ * }
+ *
+ * @phpstan-type Parsely_Posts_Stats_Response array{
+ *   data: array<string, Parsely_Post_Stats>|null,
+ *   error: Remote_API_Error|null,
+ * }
+ */
+class Admin_Columns_Parsely_Stats {
+	/**
+	 * Instance of Parsely class.
+	 *
+	 * @var Parsely
+	 */
+	private $parsely;
+
+	/**
+	 * Internal Variable.
+	 *
+	 * @var WP_Screen|null
+	 */
+	private $current_screen;
+
+	/**
+	 * Published times of visible posts.
+	 *
+	 * Analytics Endpoint don't support post_ids as param so to limit the API we will pass the
+	 * min/max publish time.
+	 *
+	 * @var string[]
+	 */
+	private $utc_published_times = array();
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Instance of Parsely class.
+	 */
+	public function __construct( Parsely $parsely ) {
+		$this->parsely = $parsely;
+	}
+
+	/**
+	 * Registers action and filter hook callbacks.
+	 *
+	 * @since 3.7.0
+	 *
+	 * @return void
+	 */
+	public function run(): void {
+		if ( ! $this->parsely->site_id_is_set() || ! $this->parsely->api_secret_is_set() ) {
+			return;
+		}
+
+		add_action( 'current_screen', array( $this, 'set_current_screen' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_parsely_stats_styles' ) );
+		add_filter( 'manage_posts_columns', array( $this, 'add_parsely_stats_column_on_list_view' ) );
+		add_action( 'manage_posts_custom_column', array( $this, 'update_published_times_and_show_placeholder' ) );
+		add_filter( 'manage_pages_columns', array( $this, 'add_parsely_stats_column_on_list_view' ) );
+		add_action( 'manage_pages_custom_column', array( $this, 'update_published_times_and_show_placeholder' ) );
+		add_action( 'admin_footer', array( $this, 'enqueue_parsely_stats_script_with_data' ) );
+	}
+
+	/**
+	 * Set current screen property.
+	 *
+	 * @since 3.7.0
+	 *
+	 * @return void
+	 */
+	public function set_current_screen(): void {
+		$this->current_screen = get_current_screen();
+	}
+
+	/**
+	 * Enqueues styles for Parse.ly Stats.
+	 *
+	 * @return void
+	 */
+	public function enqueue_parsely_stats_styles(): void {
+		if ( ! $this->is_tracked_as_post_type() ) {
+			return;
+		}
+
+		$admin_settings_asset = require_once plugin_dir_path( PARSELY_FILE ) . 'build/admin-parsely-stats.asset.php';
+		$built_assets_url     = plugin_dir_url( PARSELY_FILE ) . 'build/';
+
+		wp_enqueue_style(
+			'admin-parsely-stats-styles',
+			$built_assets_url . 'admin-parsely-stats.css',
+			$admin_settings_asset['dependencies'] ?? null,
+			$admin_settings_asset['version'] ?? Parsely::VERSION
+		);
+	}
+
+	/**
+	 * Add `Parse.ly Stats` column on admin columns.
+	 *
+	 * @param array<string, string> $columns Columns array which contain keys and labels.
+	 *
+	 * @return array<string, string>
+	 */
+	public function add_parsely_stats_column_on_list_view( array $columns ): array {
+		if ( $this->is_tracked_as_post_type() ) {
+			$columns['parsely-stats'] = 'Parse.ly Stats';
```

Comment:
Making this translatable .... this has been taken from Figma designs so as far as I know it is final but let's wait for @danielabloch also to be sure.

Reply:
`Parse.ly Stats` is good. We'd all discussed Context initially but Parse.ly Stats is better because it is clearer and includes the name of the data source (Parse.ly). 

Thanks to you both.